### ISSUE TITLE
Change adoptedStyleSheets feature detection code to be more paranoid

### ISSUE THREAD

**Issue by justinfagnani** [lit/lit-element#884]
At https://github.com/Polymer/lit-element/blob/master/src/lib/css-tag.ts#L12

instead of

```ts
export const supportsAdoptingStyleSheets =
    ('adoptedStyleSheets' in Document.prototype) &&
    ('replace' in CSSStyleSheet.prototype);
```

do something like:

```ts
export const supportsAdoptingStyleSheets =
    Array.isArray(document.adoptedStyleSheets) &&
    ('replace' in CSSStyleSheet.prototype);
```

This way if `adoptedStyleSheets` isn't an Array we fallback to creating `<style>` elements.

**Comment by web-padawan** [comment]
As long as `adoptedStyleSheets` appears to be blocked by https://github.com/WICG/construct-stylesheets/issues/45#issuecomment-577848690, how do you think the official position should eventually look like?

Should it be aligned with the one for customized builtin elements (https://github.com/Polymer/lit-element/issues/879#issuecomment-575236461) for now?

**Comment by justinfagnani** [comment]
This issue isn't about what we think the API should look like, only about making our feature detection more robust in case it changes in a backward-incompatible way.

**Comment by sorvell** [comment]
The goal here is to make our feature detection as future proof as we can with anticipated changes to the spec. However, the spec appears to be up for grabs right now so we're in a holding pattern until we know more.

**Comment by sorvell** [comment]
Closing as this is no longer relevant.
